ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Box,
+  ButtonBase,
+  ButtonBaseSize,
+  IconName,
+  Popover,
+  PopoverPosition,
+} from '../../../../component-library';
+import { TokenWithBalance } from '../asset-list';
+import {
+  getSelectedAccount,
+  getShouldHideZeroBalanceTokens,
+} from '../../../../../selectors';
+import SortControl from '../sort-control';
+import {
+  BackgroundColor,
+  BorderColor,
+  BorderStyle,
+  Display,
+  JustifyContent,
+  TextColor,
+} from '../../../../../helpers/constants/design-system';
+import { sortAssets } from '../../util/sort';
+import { useNativeTokenBalance } from '../native-token/use-native-token-balance';
+import ImportControl from '../import-control';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+
+type AssetListControlBarProps = {
+  tokenList: TokenWithBalance[];
+  setTokenList: (arg: TokenWithBalance[]) => void;
+  showTokensLinks?: boolean;
+};
+
+const AssetListControlBar = ({
+  tokenList,
+  setTokenList,
+  showTokensLinks,
+}: AssetListControlBarProps) => {
+  const [sorted, setSorted] = useState(false);
+  const controlBarRef = useRef<HTMLDivElement>(null); // Create a ref
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const tokenSortConfig = useSelector((state: any) => {
+    return state.metamask.preferences.tokenSortConfig;
```

Comment:
We can maybe create a selector for this?
` const tokenSortConfig = useSelector(getTokenSortConfig);`

and then in `selectors.js `
```
export function getTokenSortConfig(state) {
  const { tokenSortConfig } = getPreferences(state);
  return tokenSortConfig;
}
```

Reply:
I opted to destructure this from the `getPreferences` controller:

```
const { tokenSortConfig } = useSelector(getPreferences);
```

Are you fine with this, or is it best practice to create a separate selector for this single value?